### DISCUSSION TITLE
useSelector is making component rerender but the selected value does not change.

### DISCUSSION THREAD

**Discussion by tomeronen** [reduxjs/react-redux#2126] [Q&A]
Hello i would love some help with understanding a redux behavior. 
I am using react-redux version `"7.2.8"`. I have a component that is using a `useSelector` hook, the data returned from the selector does not change, but the component rerenders. When checking why, i see that a reducer inside the selector changes each time, and the number increments by one. Would really appricate help understanding what is this counter, why its incrementing and what i can do to solve it.

<img width="459" alt="image" src="https://github.com/reduxjs/react-redux/assets/45331634/5e0b90d4-7c4a-4c0e-a16a-168fb0eca0f3">

**Comment by markerikson** [comment]
Can't tell you anything without at least a CodeSandbox , Github repo, or a Replay ( https://replay.io/record-bugs ) that shows us what your actual code is and what's going on.

Also note that React re-renders recursively by default, so this component may already be re-rendering because its parent did:

- https://blog.isquaredsoftware.com/2020/05/blogged-answers-a-mostly-complete-guide-to-react-rendering-behavior/
